feat: Replace shell script with idiomatic `go generate` for code gen

iamrajiv opened the pull request:
Fixes: #81

iamrajiv commented:
@afrittoli, can you review this?

afrittoli commented on generate.go:
```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2025 The CDEvents Authors
```
NIT: 2026

afrittoli reviewed (commented):
Thanks @iamrajiv for this PR!

I think it's a good idea to add the `generate.go`, however that by itself does not replace all the functionality that is in the generate.sh script, since that scripts does the extra job of checking that no changes were made after generation, which is required in CI.

I would suggest keeping the changes you made, except for the `Makefile` one, and update the generate.sh script with the idiomatic way of invoking the generator:

```
go generate ./...
```

afrittoli commented:
@iamrajiv still looking at this one?

iamrajiv commented:
@afrittoli  can u review this again?

afrittoli commented:
@iamrajiv thanks for the updates, they look good. The PR is based on version v0.4.1  of the SDK.
Would you mind rebasing it on top of the latest main and also squashing the various commits to only 1?
Thank you!

iamrajiv committed:
feat: replace shell script with idiomatic `go generate` for code gen

Introduce a generate.go file with a //go:generate directive and update
hack/generate.sh to run `go generate ./...` instead of invoking the
generator binary directly, keeping the existing post-generation check.

Tag pkg/api/bindings_test.go and pkg/api/types_test.go with
//go:build testonly since they import the testonly-only generated
fixtures (v990/v991); this lets the untagged `go generate ./...`
package load succeed.

Fixes: #81

Signed-off-by: iamrajiv <rajivperfect007@gmail.com>

iamrajiv commented:
thanks @afrittoli done

afrittoli commented:
There is a formatting issue. If you run `make fmt` you'll see that the build tag should be placed before the copyright notice. Could you fix those please?

afrittoli commented:
BTW, this improves the readability of formatting related CI failures: https://github.com/cdevents/sdk-go/pull/142

iamrajiv committed:
fmt

Signed-off-by: iamrajiv <rajivperfect007@gmail.com>

iamrajiv commented:
@afrittoli done thanks

afrittoli reviewed (approved):
Thanks for the updates!
